library/core/src/lib.rs

```diff
@@ -210,6 +210,10 @@
 #[allow(unused_extern_crates)]
 extern crate self as core;
 
+/* The core prelude, not as all-encompassing as the std prelude */
+// The comiler expects the prelude definition to be defined before it's use statement.
```

Comment:
```suggestion
// The compiler expects the prelude definition to be defined before its use statement.
```
And same for libstd.